astropy/table/table.py

```diff
@@ -1638,6 +1716,19 @@ def add_columns(self, cols, indexes=None, copy=True, rename_duplicate=False):
                 new_indexes.insert(i, None)
                 newcols.insert(i, col)
 
+        if names is None:
+            names = ['col{}'.format(name) for name in range(len(self.columns), len(self.columns) + len(cols))]
+            for col, name in zip(cols, names):
+                if col.info.name is None:
+                    col.info.name = name
```

Comment:
@AustereCuriosity - ah, indeed, I missed that. A column in a table should indeed always have a name!  So, I guess in my example the `if col.info.parent_table is not None` could become an `elif`. Could you also add a quick comment in the code stating your reasoning?